src/main/java/Duke.java

```diff
@@ -1,10 +1,226 @@
+        System.out.println("\t Got it. I've added: this task:");
+        System.out.println("\t   " + tasks.get(tasks.size() - 1));
+        System.out.println("\t Now you have " + tasks.size() + " task(s) in the list.");
+        printHorizontalLine();
+        updateFile();
+    }
+
+    public static void deleteTask(int taskNum) {
+        if (taskNum <= 0 || taskNum > tasks.size()) {
+            System.out.println("\tInvalid task number!");
+            return;
+        }
+
+        printHorizontalLine();
+        System.out.println("\t Noted. I've removed this task:");
+        System.out.println("\t   " + tasks.remove(taskNum - 1));
+        System.out.println("\t Now you have " + tasks.size() + " task(s) in the list.");
+        printHorizontalLine();
+        updateFile();
+    }
+
+    public static boolean isEmptyDescription(String description) {
+        return ("".equals(description));
+    }
+
+    public static void markAsDone(int taskNum) {
+        if (taskNum <= 0 || taskNum > tasks.size()) {
+            System.out.println("\tInvalid task number!");
+            return;
+        } else if (tasks.get(taskNum - 1).isDone()) {
+            System.out.println("\tTask already done!");
+            return;
+        }
+        tasks.get(taskNum - 1).setDone(true);
+        printHorizontalLine();
+        System.out.println("\t Nice! I've marked this task as done:");
+        System.out.println("\t   " + tasks.get(taskNum - 1));
+        printHorizontalLine();
+        updateFile();
+    }
+
+    public static void updateFile() {
+        try {
+            writeTasksToFile();
+        } catch (IOException e) {
+            // relative path is different for testing
+            System.out.println("Check file path!");
+        }
+    }
+
+    public static void writeTasksToFile() throws IOException {
+        FileWriter file = new FileWriter(DIRECTORY_NAME + "/" + FILE_NAME);
+        for (Task t : tasks) {
+            file.write(t.getCode() + "|" + (t.isDone() ? "1" : "0") + "|" +
+                    t.getDescription() + "|" + t.getDatetime() + System.lineSeparator());
+        }
+        file.close();
+    }
+
+    public static void loadTasks() {
+        File directory = new File(DIRECTORY_NAME);
+        boolean directoryCreated = false;
+        try {
+            if (!directory.exists()) {
+                directoryCreated = directory.mkdir();
+            }
+        } catch (Exception e) {
+            e.getStackTrace();
+        }
+        // created = True -> first time running -> no need to load
+        // created = False -> directory exists OR fail to create (exists)
+        if (!directoryCreated) {
+            // handle directory exists case
+            try {
+                readTasksFromFile();
+            } catch (FileNotFoundException e) {
+                // relative path is different for testing
+                System.out.println("File not found!");
+            }
+        }
+    }
```

Comment:
Method name `loadTasks` does not coincide with function.
The function seem to be creating a directory + loading the file, rather than just loading the file.

Also, the flow of logic is a bit odd. Wouldn't it be better to check if the directory exists first, rather than trying to see if creating an directory fail?

Reply:
Noted, i will create another method for creating directory.
For the flow of logic, i actually check if the directory exists first. If not, i will create directory.
Then i check whether the directory is created successfully. If not, it means the directory exists then i read tasks from the file.